Content.Server/Doors/Components/AirlockComponent.cs

```diff
@@ -158,16 +158,35 @@ public void UpdateWiresStatus()
                 return;
             }
 
-            var powerLight = new StatusLightData(Color.Yellow, StatusLightState.On, "POWR");
-            if (PowerWiresPulsed)
+            StatusLightData powerLight;
+            if(WiresComponent != null)
             {
-                powerLight = new StatusLightData(Color.Yellow, StatusLightState.BlinkingFast, "POWR");
+                var mainPowerCut = WiresComponent.IsWireCut(Wires.MainPower);
+                var backupPowerCut = WiresComponent.IsWireCut(Wires.BackupPower);
+                var statusLightState = PowerWiresPulsed ? StatusLightState.BlinkingFast : StatusLightState.On;
+                if (mainPowerCut && backupPowerCut)
+                {
+                    powerLight = new StatusLightData(Color.Yellow, StatusLightState.Off, "POWR");
+                }
+                else if(mainPowerCut != backupPowerCut)
+                {
+                    powerLight = new StatusLightData(Color.Yellow, statusLightState, "POWR");
+                }
+                else
+                {
+                    powerLight = new StatusLightData(Color.Red, statusLightState, "POWR");
+                }
             }
-            else if (WiresComponent != null &&
-                     WiresComponent.IsWireCut(Wires.MainPower) &&
-                     WiresComponent.IsWireCut(Wires.BackupPower))
+            else
             {
-                powerLight = new StatusLightData(Color.Red, StatusLightState.On, "POWR");
+                if (PowerWiresPulsed)
+                {
+                    powerLight = new StatusLightData(Color.Yellow, StatusLightState.BlinkingFast, "POWR");
+                }
+                else
+                {
+                    powerLight = new StatusLightData(Color.Yellow, StatusLightState.On, "POWR");
+                }
             }
```

Comment:
This else here only runs if `WiresComponent == null`, which causes this whole function to have no effect anyway right?

Shouldn't the 
```cs
if (WiresComponent == null)
{
    return;
}
```
that appears later on in this function just be combined with the first `DoorComponent == null`?

Reply:
Removed the DoorComponent check, this code doesn't use the door component. Moved the Wires check to the top